pkg/framework/pod.go

```diff
@@ -180,6 +182,25 @@ func (c *ContextData) GetLogs(podName string) (string, error) {
 	return buf.String(), nil
 }
 
+func (c *ContextData) WaitForPodStatus(podName string, status v1.PodPhase, timeout time.Duration, interval time.Duration) (*v1.Pod, error) {
+	var pod *v1.Pod
+	var err error
+	ctx, cancel := context.WithTimeout(context.TODO(), timeout)
+	defer cancel()
+	for pod == nil {
```

Comment:
why a for? on a RetryWithContext, that already is doing the retry?

Reply:
Not sure why I left it there, let me try running it again and see
if this is needed, but seems like useless to me.

Reply:
@nicob87 it was probably a leftover from tests I was doing. I have removed the for statement from both pod.go and service.go. Thanks for the review.